tests/helpers/test_script.py

```diff
@@ -1484,7 +1486,8 @@ async def test_wait_template_with_utcnow_no_match(hass: HomeAssistant) -> None:
                     "variables": {"wait": {"completed": False, "remaining": None}},
                 }
             ],
-        }
+        },
+        "cancelled",
```

Comment:
The `await script_obj.async_stop()` works now and the state now correctly reflects as `cancelled`